Taunt on join

Getting this Error when I finished to connect and someone plays a taunt in that time.

[ERROR] gamemodes/ObjHunt/gamemode/cl_init.lua:119: Tried to use a NULL entity!
1. CreateSound - [C]:-1
   1. func - gamemodes/ObjHunt/gamemode/cl_init.lua:119
   2. unknown - lua/includes/modules/net.lua:32

--- a/gamemode/cl_init.lua
+++ b/gamemode/cl_init.lua
@@ -106,6 +106,8 @@ net.Receive( "Taunt Selection", function()
 	local id = net.ReadUInt( 8 )
 	local ply = player.GetByID( id )
 
+	if not IsValid( ply ) then return end
+
 	if( ply == LocalPlayer() ) then
 		ply.nextTaunt = CurTime() + ( SoundDuration( taunt ) * (100/pitch) )
 		ply.lastTaunt = CurTime()

